java/query/query-core/src/main/java/sleeper/query/core/recordretrieval/QueryExecutor.java

```diff
@@ -91,10 +91,23 @@ public void init() throws StateStoreException {
         init(Instant.now());
     }
 
+    /**
+     * Initialises a query executor with the partitions and partition to file mapping,
+     * rather than loading them from the state store.
+     *
+     * @param partitions             the partitions to initialise
+     * @param partitionToFileMapping the partition to file mapping information
+     */
     public void init(List<Partition> partitions, Map<String, List<String>> partitionToFileMapping) {
         init(partitions, partitionToFileMapping, Instant.now());
     }
 
+    /**
+     * Initialises a query executor once the next initialise time has passed.
```

Comment:
This seems to suggest it might wait for the initialise time to pass. We could replace "once" with "if"?